python/tests/test_topology.py

```diff
@@ -45,43 +45,52 @@ def slice(
+        stop = [ts.sequence_length]
+    zipRanges = list(zip(start, stop))
+    # not sure how best to reset coords if slicing multiple intervals
+    # so currently defaulting to False
+    if len(zipRanges) > 1:
+        reset_coordinates = False
+    if any(x < 0 for x in start) or any(x[1] <= x[0] for x in zipRanges) \
+       or any(x > ts.sequence_length for x in stop):
         raise ValueError("Slice bounds must be within the existing tree sequence")
     tables = ts.dump_tables()
     tables.edges.clear()
     tables.sites.clear()
     tables.mutations.clear()
     for edge in ts.edges():
-        if edge.right <= start or edge.left >= stop:
-            # This edge is outside the sliced area - do not include it
-            continue
-        if reset_coordinates:
-            tables.edges.add_row(
-                max(start, edge.left) - start, min(stop, edge.right) - start,
-                edge.parent, edge.child)
-        else:
-            tables.edges.add_row(
-                max(start, edge.left), min(stop, edge.right),
-                edge.parent, edge.child)
-    for site in ts.sites():
-        if start <= site.position < stop:
+        for aRange in zipRanges:
+            if edge.right <= aRange[0] or edge.left >= aRange[1]:
+                # This edge is outside the sliced area - do not include it
+                continue
             if reset_coordinates:
-                site_id = tables.sites.add_row(
-                    site.position - start, site.ancestral_state, site.metadata)
+                tables.edges.add_row(
+                    max(aRange[0], edge.left) - aRange[0],
+                    min(aRange[1], edge.right) - aRange[0],
+                    edge.parent, edge.child)
             else:
-                site_id = tables.sites.add_row(
-                    site.position, site.ancestral_state, site.metadata)
-            for m in site.mutations:
-                tables.mutations.add_row(
-                    site_id, m.node, m.derived_state, m.parent, m.metadata)
+                tables.edges.add_row(
+                    max(aRange[0], edge.left), min(aRange[1], edge.right),
+                    edge.parent, edge.child)
+    for site in ts.sites():
+        for aRange in zipRanges:
+            if aRange[0] <= site.position < aRange[1]:
+                if reset_coordinates:
+                    site_id = tables.sites.add_row(
+                        site.position - aRange[0], site.ancestral_state, site.metadata)
+                else:
+                    site_id = tables.sites.add_row(
+                        site.position, site.ancestral_state, site.metadata)
+                for m in site.mutations:
+                    tables.mutations.add_row(
+                        site_id, m.node, m.derived_state, m.parent, m.metadata)
     if reset_coordinates:
-        tables.sequence_length = stop - start
+        tables.sequence_length = stop[0] - start[0]  # this is not ideal
```

Comment:
This should be `sum([y - x for x, y in zip(start, stop)])`, if we cut out all between-interval bits?